agent/dns_test.go

```diff
@@ -2997,6 +2997,163 @@ func testDNSServiceLookupResponseLimits(t *testing.T, answerLimit int, qType uin
 	return true, nil
 }
 
+func testDNSServiceLookupResponseARecordLimits(t *testing.T, generateNumNodes int, aRecordLimit int, qType uint16,
```

Comment:
Can just have this return an error instead of both an error and bool, and check whether an error occurred.

Also, can you rename this to checkDNSService...

Reply:
@preetapan DONE in next patch